From a09feee6d7768d0e91e9359bd73e461ab3b0e8c7 Mon Sep 17 00:00:00 2001
From: Florent Revest <revestflo@gmail.com>
Date: Wed, 10 May 2017 20:26:55 +0200
Subject: [PATCH] [systemd] Use a default timeout of 2sec to StartUnit and
 StopUnit

On my devices, I noticed that adding udhcpd-developer-mode.ini to
/etc/usb-moded/run/ would totally prevent USB from working. It turned
out that usb-moded was trying to call StopUnit on udhcpd.service and
the blocking call would never stop. I had to introduce this timeout to
make sure usb-moded would start properly. It may break the starting and
stopping time of other services though.
---
 src/usb_moded-systemd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/usb_moded-systemd.c b/src/usb_moded-systemd.c
index 4a27cfb..a59ab69 100644
--- a/src/usb_moded-systemd.c
+++ b/src/usb_moded-systemd.c
@@ -84,7 +84,7 @@ int systemd_control_service(const char *name, const char *method)
         	dbus_message_unref(msg);	
 		goto quit;
 	}
-	reply = dbus_connection_send_with_reply_and_block(bus, msg, -1, &error);
+	reply = dbus_connection_send_with_reply_and_block(bus, msg, 2000, &error);
 	if(reply)
  	{
 		dbus_message_unref(reply);
-- 
2.11.0

